bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/FreeAtHomeSystemHandlerFactory.java

```diff
@@ -0,0 +1,127 @@
+import org.openhab.binding.freeathomesystem.internal.type.FreeAtHomeChannelGroupTypeProvider;
+import org.openhab.binding.freeathomesystem.internal.type.FreeAtHomeChannelTypeProvider;
+import org.openhab.binding.freeathomesystem.internal.type.FreeAtHomeThingTypeProvider;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.openhab.core.thing.type.ThingType;
+import org.openhab.core.thing.type.ThingTypeBuilder;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeSystemHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Andras Uhrin - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.freeathomesystem", service = ThingHandlerFactory.class)
+public class FreeAtHomeSystemHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeSystemHandlerFactory.class);
+
+    private @NonNullByDefault({}) HttpClient httpClient;
+    private final FreeAtHomeThingTypeProvider thingTypeProvider;
+    private final FreeAtHomeChannelTypeProvider channelTypeProvider;
+    private final FreeAtHomeChannelGroupTypeProvider channelGroupsTypeProvider;
+
+    private void generateThingTypes() {
+        String label = "free-at-home-device";
+        String description = String.format("Generic free@home device");
+
+        List<String> supportedBridgeTypeUids = new ArrayList<>();
+        supportedBridgeTypeUids.add(BRIDGE_TYPE_UID.toString());
+        ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, FREEATHOMEDEVICE_TYPE_ID);
+
+        Map<String, String> properties = new HashMap<>();
+        properties.put(Thing.PROPERTY_VENDOR, "Busch&Jaeger - ABB");
+        properties.put(Thing.PROPERTY_MODEL_ID, "free@home Device");
+
+        URI configDescriptionURI;
+        try {
+            configDescriptionURI = new URI("thing-type:freeathomesystem:free-at-home-device");
+
+            ThingType thingType = ThingTypeBuilder.instance(thingTypeUID, label)
+                    .withSupportedBridgeTypeUIDs(supportedBridgeTypeUids).withDescription(description)
+                    .withProperties(properties).withConfigDescriptionURI(configDescriptionURI).build();
+
+            this.thingTypeProvider.addThingType(thingType);
+        } catch (URISyntaxException e) {
+            logger.debug("Exception during creating config description URI");
```

Comment:
I guess the binding will not work then, so probably error could be justified? See https://www.openhab.org/docs/developer/guidelines.html#f-logging
```suggestion
            logger.error("Exception during creating config description URI");
```